### ISSUE TITLE
Allow "next mission in chain" to be displayed on the map

### ISSUE THREAD

**Issue by danaris** [endless-sky/endless-sky#9482]
### Problem Description

One of the more frustrating experiences in ES is not knowing what to do next, either because you just finished a mission and it didn't make it entirely clear or because you left a mission chain partway through and when you tried to go back to it, there was no easy indication of where to do so.

### Related Issue Links

#7449 — proposes adding an explicit concept of "storylines"; this could definitely interact with that

### Desired Solution

It would be extremely helpful, for missions that are part of a longer chain (and _especially_ for the main plot) to have some sort of marker be added to the map where you can pick the up when the previous mission is complete.

The three most obvious ways I can see to do this are:

1. Simply add the ability for any mission to add a Marker to the map
2. Create a new type of data object, Marker, that has `to display` conditions whose format mirrors `to offer` conditions and location options that match the `source` options for Missions
3. Add a new flag for missions, something like `mark on available`

The markers associated with this should be a different color than those currently used for missions and jobs (or possibly multiple different colors, to be able to differentiate between main plot and other types of storyline missions). Ideally, it should also be possible to have some sort of indicator on the Spaceport for the marked planet (or, for future missions that actually use the ability to start from different planetary locations, on the buttons for those locations).

It would also be a good idea to give the player the ability to disable and re-enable these at will—either all at once, or one by one.

### Alternative Approaches

It could also be possible to add to the Logbook somewhere an indication that there's an expectation of some continuation at <Systemname>, if it is considered undesirable to have these on the map. However, this seems like it would require basically the same amount of work to make the information available.

**Comment by Hecter94** [comment]
I think the alternative approach you've given is the better approach here. 

It's often been discussed that we want to give the player freedom in approaching the world of Endless Sky, meaning not directly pointing them toward mission chains, not applying too much direction within them, and not making the individual chains feel too forced.
Players should feel more like they are a part of this world and less like they have a log full of quests to complete, with markers pointing them toward exactly where they need to go to start/continue/finish each mission. 

This comes with the understanding that players will often miss things, they may not always get every mission, or may completely miss certain storylines, and that is okay; it's part of each player's unique journey through the world of Endless Sky.

With that said, there is a specific scenario, which you have alluded to above, which is less of an intentional design choice and more of a limitation of the existing mechanics, which are cases where, to continue a storyline, you are required to pick up the next part of the mission by visiting a spaceport, typically the same spaceport as the player is currently on, and while they are told to visit the spaceport, if the player does or can not at the moment and leaves with the intent to return, it's often not possible to find anywhere in-game that would remind the player which spaceport they need to visit when they are ready to return to that mission chain. 

The best option for rectifying such a situation would be to provide the player with a means of obtaining an in-universe reminder of where to pick up the next quest in the chain; the most logical place for this would be the player's logbook. 
A simple, "`npc` asked to meet me to meet them at the spaceport on `planet` when I'm ready to continue assisting them.", written to the logbook whenever a quest has a break that requires visiting the spaceport would allow an easy in-universe reminder of where to go next, without populating the game map with dozens of markers pointing the player exactly where to go. 

If we're concerned about the discoverability of such a system, then we could go with a system similar to what is mentioned in https://github.com/endless-sky/endless-sky/issues/7449#issuecomment-1474932587:
> perhaps instead of only having missions create logbook entries for the year they happened in, we could have individual tabs for individual storylines, with entries that clearly state when a storyline has ended.

This would allow a lost player to quickly scan through their logbook, viewing each storyline and able to see at a glance whether that storyline is completed, and if it is not completed, have a written in-universe reminder of where to go to pick that storyline back up again as the final note in their logbook for that storyline.

**Comment by Amazinite** [comment]
Some more related stuff:
#4803
#2712
#2342

**Comment by Amazinite** [comment]
I was going to open a new issue, but I found this one while looking for related issues and figured it was close enough to the same issue. Here's the issue description I'd started writing up.

<hr/>

Right now if you're playing through a storyline and end up reaching a mission that is blocked due to you not having enough space for the mission, you end up in a state where you have no active missions (for that storyline) and need to remember where a mission was supposed to be offered while you go off and make space for the mission. Depending on how difficult it is for you to meet the requirements for a mission, this might take some time for a player, by which point they might have forgotten exactly which planet the mission is supposed to be offered on.

<hr/>

Resolving this issue would also mean resolving the case I described.

**Comment by Amazinite** [comment]
> * Simply add the ability for any mission to add a Marker to the map
> * Create a new type of data object, Marker, that has `to display` conditions whose format mirrors `to offer` conditions and location options that match the `source` options for Missions
> * Add a new flag for missions, something like `mark on available`

I don't think we should do the last one. Right now, the game only looks for which missions are available at your current location. I simply can't see any way how we'd be able to have missions be tagged `mark on available` and have us somehow figure out whenever you land where every available mission is across every possible offer location.

I think the first option is covered by #9827, but I don't think that alone helps us, since it would mean that for every mission that we want to add a reminder to or display that it's available, we'd need an entirely separate mission for the purpose of adding the marker. Although, having a `marker` root node would still be the exact same thing of needing a separate root node for the purposes of creating the ping. 🤔
